ibind/base/rest_client.py

```diff
@@ -147,8 +187,21 @@ def request(self, method: str, endpoint: str, attempt: int = 0, log: bool = True
             Exception: For any other errors that occur during the request.
 
         """
+
+        # TODO: raise error if use_oauth=False, but live_session_token=None
```

Comment:
Do I understand correctly that you meant `if use_oauth=True` here, rather than `False`?

Also, note that RestClient doesn't have access to `self._use_oauth`, only IbkrClient does. Hence, we'll have to tackle that differently - possibly in `IbkrClient.get_headers()`?

Reply:
I haven't check that both methods work simply by changing use_oauth from True to False. I'm guessing there may be an error here.

I added use_oauth to ibkr_client ```super().__init__(url=url, use_oauth=use_oauth,cacert=cacert, timeout=timeout, max_retries=max_retries)```. There is probably a better method, but this worked.

Reply:
Right, I understand why you did it this way. 

Putting my software designer hat on for a second, let me try to explain why we cannot use that variable here: 

The RestClient class is an abstract class. The IbkrClient is its concrete implementation. The `use_oauth` member field is declared within the IbkrClient and therefore only available within it. RestClient, as the 'parent' class, also _techincally_ has access to any variables defined in its 'children', but from design perspective it doesn't know about it, and hence shouldn't use it. It will work if you run it, but it's a design bug.

Hypothetically if there was another IBKR rest class, let's say IbkrClientNew, that uses a different API that doesn't even expose OAuth, and therefore it doesn't define `use_oauth`, this line would cause an error. I know this is hypothetical at the moment, but I do envision decoupling RestClient and WsClient base classes to their own package some day, hence I try to think of these things already.

There are ways we can get around this, and my introduction of 'base_url' parameter in `request` method was one of such moves. For this TODO, we could for example do this check in `IbkrClient.get_header` and avoid this design bug, since we'd be checking for `use_oauth` within `IbkrClient` that has access to it.

----

Edit1: I've just realised you've added `use_oauth` to RestClient. If that was the case then the above argument is invalid. I write this with the assumption that `use_oauth` is only defined in IbkrClient as we've originally implemented it.

Reply:
That's correct, ```use_oauth``` is only defined in IbkrClient as originally designed.